sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala

```diff
@@ -71,19 +75,39 @@ case class ShowCreateTableExec(
       builder: StringBuilder,
       tableOptions: Map[String, String]): Unit = {
     if (tableOptions.nonEmpty) {
-      val props = tableOptions.toSeq.sortBy(_._1).map { case (key, value) =>
-        s"'${escapeSingleQuotedString(key)}' = '${escapeSingleQuotedString(value)}'"
+      val props = SQLConf.get.redactOptions(tableOptions).toSeq.sortBy(_._1).map {
+        case (key, value) =>
+          s"'${escapeSingleQuotedString(key)}' = '${escapeSingleQuotedString(value)}'"
       }
       builder ++= "OPTIONS "
       builder ++= concatByMultiLines(props)
     }
   }
 
   private def showTablePartitioning(table: Table, builder: StringBuilder): Unit = {
-    if (!table.partitioning.isEmpty) {
-      val transforms = new ArrayBuffer[String]
-      table.partitioning.foreach(t => transforms += t.describe())
-      builder ++= s"PARTITIONED BY ${transforms.mkString("(", ", ", ")")}\n"
+    table match {
+      case V1Table(t) =>
```

Comment:
We don't need to special-case v1 table. We should recognize these special partition transforms (`bucket` and `sorted_bucket`) and generate the CLUSTERED BY syntax for them.

@huaxingao have we unified this part? We discussed this before and the parser should generate partition transforms for bucket spec already, instead of `BucketSpec`

Reply:
Haven't changed this yet. Remember that 
`c.partitioning ++ c.tableSpec.bucketSpec.map(_.asTransform)` didn't work inside `AstBuilder` because sortColumnNames needed to be empty.  Now it should work. I can have a follow up to fix this.